Tests/FluentAssertions.Shared.Specs/GenericCollectionAssertionsSpecs.cs

```diff
@@ -34,6 +34,444 @@ public void When_asserting_equality_with_a_collection_built_from_params_argument
+            Action assert = () =>
+            {
+                actualDates.Should()
+                    .BeEquivalentTo(expectedDates, (actual, expected) => actual.Should().BeSameDateAs(expected));
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            //Assert
+            //-----------------------------------------------------------------------------------------------------------
+            assert.ShouldThrow<AssertFailedException>();
+        }
+
+        [TestMethod]
+        public void Check_asserting_equivallency_by_assert_method_should_throw_exception_when_expecting_collection_is_bigger()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            //Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            List<DateTime> actualDates = new List<DateTime>
+            {
+                DateTime.Today,
+                DateTime.Today.AddDays(2)
+            };
+
+            List<DateTime> expectedDates = new List<DateTime>
+            {
+                DateTime.Now,
+                DateTime.Now.AddDays(2),
+                DateTime.Today.AddDays(2)
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            //Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action assert = () =>
+            {
+                actualDates.Should()
+                    .BeEquivalentTo(expectedDates, (actual, expected) => actual.Should().BeSameDateAs(expected));
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            //Assert
+            //-----------------------------------------------------------------------------------------------------------
+            assert.ShouldThrow<AssertFailedException>();
+        }
+
+        [TestMethod]
+        public void Check_asserting_equivallency_should_throw_indicative_missing_items_message()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            //Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            const string BECAUSE = "there are some missing members";
+            List<DateTime> missingItems = new List<DateTime>
+            {
+                DateTime.Today.AddDays(2),
+                DateTime.Today.AddDays(7)
+            };
+            String missingItemsMessage = String.Join(", ", missingItems.Select(it => "<" + it.ToString("yyyy-MM-dd") + ">"));
```

Comment:
Do not duplicate library logic in tests. Please use constant values for actual and expected values and manually prepare expected message